Sprint_6

DmitriyVOsipov opened the pull request:
Sprint_6

DmitriyVOsipov committed:
add jacoco report

DmitriyVOsipov committed:
result Jacoco

egorvas commented on src/main/java/com/example/Animal.java:
```diff
@@ -0,0 +1,20 @@
+package com.example;
+import java.util.List;
+
+public class Animal {
```
Ты неверно сделал пул реквест. Твой код должен мержится в бранч с исходными классами, чтобы было видно какие именно ты правки внес. А у тебя пул реквест в пустой бранч, поэтому здесь видно все файлы. Тебе нужно ознакомится с инструкцией по подаче и попробовать еще раз.

DmitriyVOsipov replied:
Я удалил исходный проект с гита :-( . Пришлось создавать новый с нуля. На компьютере остался готовый проект. По этому я запушил готовый проект и не видно изменений.
Уточнял в группе у наставника, мне подсказали сделать проект с нуля. Я так и сделал.
Если бы было можно заново его создать, я бы сначала запушил исходные файлы в ветку main, потом запушил измененный код develop->develop  и сделал пул реквест в main.
Но исходные файлы уже изменены.

egorvas commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,45 @@
+		Assert.assertEquals("Мяу", cat.getSound());
+	}
+
+	@Test(expected = NullPointerException.class)
```
В случае если ожидаемым результатом теста является эксепшен я рекомендую проверять не только что он произошел, но и текст ошибки(если он есть). Я не считаю это ошибкой, можно не исправлять, но по желанию можно попробовать добавить в тест проверку текста.

egorvas commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,54 @@
+		} catch (Exception e) {
+			String message = "Неизвестный вид животного, используйте значение Травоядное или Хищник";
+			Assert.assertEquals(message, e.getMessage());
+			throw e;
```
Вот это довольно странная конструкция. Тут тете нужно поискать способ получше для проверки.
Рекомендую убрать expected = Exception.class и добавить в секцию try проверку после feline.getFood("null"), которая будет всегда падать.

DmitriyVOsipov replied:
Удалю, проверку сделаю.

egorvas commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,79 @@
+	@Mock
+	Feline feline;
+
+	@Parameterized.Parameters(name = "Пол = {0}, Грива = {1}")
```
Тут довольно грубая ошибка. В случае если произойдет неожиданный эксепшен в позитивном сценарии, то твой тест не упадет попав в catch.
В случае отсутствия эксепшена при негативном сценарии, твой тест также не упадет, так как у тебя вообще нет ассерта на это.
Тут нужно разделить на два теста - параметризованный с позитивными сценариями и отдельный на эксепшен. Причем нужно обязательно проверять, что эксепшен произошел.

DmitriyVOsipov replied:
Разделю на 2 теста

egorvas commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,79 @@
+		try {
+			Lion lion = new Lion(sex, feline);
+			List<String> expectedList = List.of("Животные", "Птицы", "Рыба");
+			Mockito.when(lion.getFood()).thenReturn(expectedList);
```
Это очень грубая ошибка. Ты сначала замокала функцию, а потом ее же проверяешь. По сути твой тест можно переписать как assertEquals(List.of("Животные", "Птицы", "Рыба"), List.of("Животные", "Птицы", "Рыба"))
Мало того что это просто бесполезно, так ко всему прочему вводит в заблуждение в отчете о покрытии, в реальности твоя функция не проверяется.
Тут можно мокать только функции feline.

DmitriyVOsipov replied:
Учту при составлении двух тестов.

DmitriyVOsipov committed:
Fix FeelineTest
Удалил тест "expected = Exception.class".
Добавил  "try после feline.getFood("null")""

DmitriyVOsipov committed:
Fix LionTest
Разбил на 2 теста.
На параметризованный и отдельно на эксепшен.
Мок поверяемой функции убрал.

DmitriyVOsipov committed:
Jacoco результат тестов

DmitriyVOsipov committed:
Коммит проекта
